### PR TITLE
Fixed assert breaking tests

### PR DESCRIPTION
**Pull request by mikeger**
- Replaced the assert with `ZMLogError`.

Idea for the future improvement: 
Create the precondition method that checks the condition and fails the test but crash on production.

### CODE REVIEW

**Comment by typfel** [comment]
Most transcoders actually check if they are in the correct sync phase before finishing/failing it. `ZMLastUpdateEventIDTranscoder` doesn't do this and that's why it asserts.

**Comment by typfel** [comment]
`line: 133` in `ZMLastUpdateEventIDTranscoder ` should we wrapped in a `if (self.isSyncing) {}` to prevent this.

**Comment by mikeger** [comment]
@typfel ok, I would try to fix it that way then.

**Comment by mikeger** [comment]
@typfel updated. Waiting for CircleCI.

**Comment by mikeger** [comment]
Works!

File: Source/Synchronization/Transcoders/ZMLastUpdateEventIDTranscoder.m [hunk 1]
```diff
@@ -139,7 +141,10 @@ - (void)didReceiveResponse:(ZMTransportResponse *)response forSingleRequest:(ZMS
         self.lastUpdateEventID = lastNotificationID;
         if (status.currentSyncPhase == self.expectedSyncPhase) {
             [status updateLastUpdateEventIDWithEventID:lastNotificationID];
-            [status finishCurrentSyncPhaseWithPhase:self.expectedSyncPhase];
+            
+            if (status.isSyncing) {
```
Comment (typfel) [inline Source/Synchronization/Transcoders/ZMLastUpdateEventIDTranscoder.m]: this is the same as `status.currentSyncPhase == self.expectedSyncPhase` in the outer `if` statement

**Pull request merged**